crossgen.sh improvements

ellismg opened the pull request:
This PR has two changes:

1) Don't crossgen a file if the ni is already present.  This will address the segmentation faults we see when we try to crossgen System.Private.Corelib.dll (the .ni.dll already exists, it's in the package).
2) If `NUGET_PACKAGES` is set in the environment, respect that variable as the location to restore packages to.

I'd like some sort of knob to allow me to restore packages to a shared location as part of the build from source work, and `NUGET_PACKAGES` seemed like the obvious choice, since the tooling understands that by default.

weshaggard commented on src/Microsoft.DotNet.Build.Tasks/PackageFiles/crossgen.sh:
```diff
@@ -54,13 +54,17 @@ crossgen_single()
-            __outname="${__file/.dll/.ni.dll}"
-            __outname="${__outname/.exe/.ni.exe}"
-            echo "$__file -> $__outname"
+        if [[ ($__file == *.dll && -e ${__file/.dll/.ni.dll}) || ($__file == *.exe && -e ${__file/.exe/.ni.exe}) ]]; then
```
This isn't going to help System.Private.CoreLib case because we no longer name the file ni.dll it is just .dll

ellismg replied:
It helped locally with the latest buildtools.  Do you know when that change went in?

weshaggard replied:
https://github.com/dotnet/coreclr/pull/11359

weshaggard commented on src/Microsoft.DotNet.Build.Tasks/PackageFiles/crossgen.sh:
```diff
@@ -78,7 +82,7 @@ __targetDir=$1
 __toolsDir=$__scriptpath/../Tools
 __dotnet=$__toolsDir/dotnetcli/dotnet
-__packagesDir=$__scriptpath/../packages
+__packagesDir="${NUGET_PACKAGES:-${__scriptpath}/../packages}"
```
Did we ever end up putting crossgen into the shared framework? That would make this script a lot simpler.

ellismg replied:
It is in Microsoft.NETCore.App now, in the tools folder.

weshaggard replied:
In that case does it make since to stop hardcoding the coreclr package version and instead use the shared framework version to restore Microsoft.NETCore.App instead?

ellismg replied:
I can fix that after 2.0

weshaggard replied:
sounds good. Might be worth filing an issue.

ellismg commented:
@weshaggard I added an explicit check to not crossgen System.Private.Corelib.dll.

I also added a commit that makes it possible to specify CommitHash and VersionSeedDate properties explicitly instead of always shelling out to `git` (which is useful for the build for source stuff).

weshaggard commented on src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets:
```diff
@@ -285,7 +291,12 @@
+      <LatestDateCommandExitCode>0</LatestDateCommandExitCode>
+    </PropertyGroup>
+
+
```
nit: extra blank line
